src/main/java/abstracts/Car.java

```diff
@@ -0,0 +1,217 @@
+    }
+
+    public void setActualGear(int actualGear) {
+        this.actualGear = actualGear;
+    }
+
+    public float getLoadCapacity() {
+        return loadCapacity;
+    }
+
+    public void setLoadCapacity(float loadCapacity) {
+        this.loadCapacity = loadCapacity;
+    }
+
+    public int getNumberOfDoors() {
+        return numberOfDoors;
+    }
+
+    public void setNumberOfDoors(int numberOfDoors) {
+        this.numberOfDoors = numberOfDoors;
+    }
+
+    public int getMaxNumberOfOccupants() {
+        return maxNumberOfOccupants;
+    }
+
+    public void setMaxNumberOfOccupants(int maxNumberOfOccupants) {
+        this.maxNumberOfOccupants = maxNumberOfOccupants;
+    }
+
+
+    /**
+     *Check if the car is already full and try to add the occupant. If the occupant is already in the car it gets error.
+     *
+     * @param name name of occupant that will enter in the motorcycle
+     */
+    public void addOccupant(String name){
+
+        if (occupantsNames.size() < maxNumberOfOccupants){
+            try {
+                occupantsNames.add(name);
+            }catch (Exception e){
+                logger.error("Fail to insert new occupant. Already in car. " + e);
+            }
+        }else {
+            logger.error("The car is full!");
+        }
+    }
+
+    /**
+     * Check if has at least one occupant and try to remove. If the ocuppant is not in the motorcycle, show an error.
+     * @param name name of occupant that will exit in the motorcycle
+     */
+    public void removeOccupant(String name){
+
+        if(occupantsNames.size() > 0){
+            try{
+                occupantsNames.remove(name);
+            }catch (Exception e){
+                logger.error("Fail to remove the occupant. Not in car. " + e);
+            }
+        }else {
+            logger.error("There is no more occupants in the car");
+        }
+    }
+
+
+    @Override
+    public String toString() {
+        return "Car{" +
+                "fuel=" + fuel +
+                ", numberOfGears=" + numberOfGears +
+                ", actualGear=" + actualGear +
+                ", loadCapacity=" + loadCapacity +
+                ", numberOfDoors=" + numberOfDoors +
+                ", maxNumberOfOccupants=" + maxNumberOfOccupants +
+                ", occupantsNames=" + occupantsNames +
+                '}';
+    }
+
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Car car = (Car) o;
+        return Float.compare(car.fuel, fuel) == 0 &&
+                numberOfGears == car.numberOfGears &&
+                actualGear == car.actualGear &&
+                Float.compare(car.loadCapacity, loadCapacity) == 0 &&
+                numberOfDoors == car.numberOfDoors &&
+                maxNumberOfOccupants == car.maxNumberOfOccupants &&
+                Objects.equals(occupantsNames, car.occupantsNames);
+    }
```

Comment:
Your equals methods look pretty good but think about implementing some kind of id attribute for the Vehicles like model, plate or registration number.